lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -397,6 +397,8 @@ class ClassMetadataInfo implements ClassMetadata
      * Whether a unique constraint should be generated for the column.
      *
      * @var array
+     *
+     * @psalm-var array<string, array{type: string, fieldName: string, columnName: string, inherited: class-string}>
```

Comment:
/cc @orklah `nullable` is missing here. Also, all the array keys are supposed to be optional but here they're required...